Replace gnome-keyring with pass/pass-secret-service

jzetterman opened the pull request:
This PR addresses issue #352 by replacing gnome-keyring with [pass](https://www.passwordstore.org/) and pass-secret-service. In testing multiple options, Gnome Keyring, Kwallet, and KeePassXC all presented significant challenges to automating the creation of a keyring that could be provisioned from the command line in a way that didn't require any user interaction. Pass was the only solution I found capable of meeting the goal of no user interaction in the provisioning of a passwordless keystore that apps such as 1Password can use for storing secrets such as MFA tokens. 

I've tested this on a fresh VM inside Proxmox and on an older Thinkpad X1 Yoga and it worked flawlessly in both scenarios with saving 1Password MFA tokens. That said, this is my first PR ever (in any capacity) and I am not a developer by trade, but an IT systems engineer/manager for the last several years. If there is room for improvement I am totally open to it. For example, I made an educated guess about where the logic inside keyring.sh should go and considered a few options. Making it it's own file seemed like the most logical way to handle it. I'm also not sure if this warrants a migration and haven't developed one. My thought is that existing Omarchy users might not want their current keyring solution to be replaced. 

Thanks for your consideration!

jzetterman commented on install/config/keyring.sh:
```diff
@@ -0,0 +1,45 @@
+EOF
+    fi
+
+    gpg --batch --generate-key gpg-keygen.conf
```
Just reviewed the man page docs and you're right. I'll remove that.

jzetterman replied:
Upon further review, it's not well documented, but it does appear to be a requirement for unattended key generation: 
https://man.archlinux.org/man/gpg.1.en#Unattended_key_generation

jzetterman commented on install/config/keyring.sh:
```diff
@@ -0,0 +1,45 @@
+    gpg --batch --generate-key gpg-keygen.conf
+
+    # Get the GPG key ID
+    GPG_ID=$(gpg --list-keys --keyid-format LONG | grep pub | awk '{print $2}' | cut -d'/' -f2)
```
Ok, I'm playing with how to do this.

jzetterman replied:
Alright, I have a method for pulling the keyring UID from the output of the generation event, which should be more reliable (at least until GPG ever updates the output). I'll update the logic.

jzetterman replied:
@ryanrhughes sorry for the mess I made, I was fighting Git for a bit there. The new logic is now reflecting. The main change is it parses the output after running the GPG generate key command from stdout and pulls out the 16 character UID from the output and uses that specifically when running `pass init`. I've tested on my Arch dev VM in my Proxmox dev environment and everything worked as expected.

rafaelbernard commented on boot.sh:
You can run `git checkout master boot.sh`, then commit it. It will roll back the file to the current master version.

roelandmoors commented:
I would also install this during a 'bare' installation. Seems essential to me.

jzetterman commented:
@ryanrhughes I believe I've addressed all of your comments. The one thing I need some guidance on is how to develop a migration. I'm not sure how those are being produced and what the numbering scheme means, for example 1754568612.sh, etc... I can work through the logic, though I think it will be fairly complex or at least has the potential to be, but we can work through that. Thanks in advance!

jzetterman commented:
> I think this is just a unix timestamp: https://www.unixtimestamp.com/

That makes sense!

> 
> Is the idea to remove gnome-keyring in a migration and install pass/pass-secret-service? Or only install pass/pass-secret-service if gnome-keyring is not installed.
> 
> The second options is safer for now, but in the long run I prefer the first option. Then all Omarchy installs have the same keyring system.

My thought was that it made more sense to go with Option 2 in a migration. I think Option 1 is achieved with time as gnome-keyring is no longer installed by default it will fade away as people refresh their machines. The last thing any user would want is for their stored tokens to be deleted by Omarchy during an upgrade. 

On a more philosophical note, I don't think Omarchy should be telling people what keyring to use if they make the active choice to use another one. I see it as a baseline to be customized on top of as needed and I think that's @dhh's philosophy as well.

jzetterman commented:
> @jzetterman for future `omarchy-dev-add-migration` will create a migration for you but the naming is just the current epoch.

Got it, thanks for pointing me in the right direction. That seems simple. Anything special about the migration file when I create it? Looking through the existing migrations I'm noticing the lack of the shebang, but otherwise looks like normal bash scripts. I also noticed that 1752082381.sh installs gnome-keyring. We may need to remove that one. 

> For migration, I'd suggest we just look at the user's keyring. If they don't have any, safe to say we can ditch Gnome and setup the better option for them without intervention.
> 
> Otherwise, we present a Gum confirmation asking them what they want us to do for them.

Tentatively, I'm planning on checking for the presence of gnome-keyring, kwallet, and keepassxc processes and packages along with checking their usual keyring locations. If located I'll present the Gum confirmation with a scary warning. Yay is still down, so I can't do a lot of testing at the moment, but I'll work on this over the course of the next few days.

iantriggs commented:
Would this mean passwords would be stored in plain text? If so I'd much prefer that a login manager was just added to Omarchy.

jzetterman commented on install/config/keyring.sh:
```diff
@@ -0,0 +1,37 @@
+yay -S --noconfirm --needed pass pass-secret-service-bin
+
+# STEP 1: Generate GPG key non-interactively and capture output
+if [ -z "$OMARCHY_USER_NAME" ] || [ -z "$OMARCHY_USER_EMAIL" ]; then
```
Those are used to generate the GPG key, so my thought was to pull them from the Git user and email to ensure uniqueness and only use Omarchy and omarchy@keyring.local when those didn't exist to ensure uniqueness of users keys. If we use the same name and email for everyone then I believe (though I haven't tested to say for certain) that every Omarchy users GPG key would be identical. That may not be optimal. I'll do some tests to see if that's the case.

jzetterman commented:
@ryanrhughes I've been working on the migration, but ran into a roadblock with getting pass to work after remove gnome-keyring. I also didn't have as much time as I'd have liked to work on it this week. I'll be spending my evening tonight digging in to see if I can figure that out.

jzetterman commented:
@ryanrhughes the migration for this change is ready. I wasn't sure how to handle "migrations/1752082381.sh" so all I did was comment everything out in that file. That migration conflicts with this change and needs to be removed or archived in whatever way that can be achieved. 

I tested the migration pretty extensively and occasionally I was running into `status 429: Rate limit reached` from the AUR. Today is the first time I've ever seen that one and my guess is that they're doing this to combat the ongoing DDoS attack. I tried playing around with capturing errors from yay, but it seems that yay sends an exit code of 0, even when there are errors. At the end of the day I figured that if the AUR isn't working then it's going to be a problem for more than just this migration so I didn't end up dealing with that case. Plus I saw DHH announced on X that the migration to an Omarchy-specific package repository was complete so it'd probably be better to move `pass-secret-service-bin` (`pass` comes from Arch's extra repo and wasn't affected) there anyway. 

On the gum prompt, I defaulted to no, that may not be the correct thing to do. I default to being conservative with decisions like that though, but I can change it if you feel like it should default to yes. In order to get it to default to no I had to use gum in a slightly different way than it's used elsewhere which may also be problematic. 
<img width="666" height="452" alt="image" src="https://github.com/user-attachments/assets/2c7527bd-26ec-4983-8644-bed2e9c6614c" />

arcbjorn reviewed (changes requested):
The current keyring implementation initializes a single password store at `~/.password-store/` with an auto-generated GPG key. This creates conflicts with personal pass store:
  1. Omarchy Install → ~/.password-store/ (system GPG key)
  2. User reinitializes pass with their own GPG key → System keyring breaks
  3. User runs `pass git init` in `~/.password-store/` → System secrets get committed to their personal repo
 
  **Better flow**:
  Omarchy Install → ~/.omarchy-keyring/ (system GPG key)
  User later → ~/.password-store/ (personal GPG key) → No conflicts
  User later → git init ~/.password-store/ → Only personal secrets

  Separate the keyring stores entirely:
  1. System keyring uses ~/.omarchy-keyring/ with Omarchy-generated GPG key
  2. pass-secret-service configured to use system keyring via PASSWORD_STORE_DIR
  
  This basically provides clean separation, no conflicts, git safety, better Omarchy security and user freedom. Please, check my comments with improvement suggestions @jzetterman

ryanrhughes commented:
After spending some time with this solution, I've decided I think the better way to go is what I've done in #1860. It accomplishes the same thing, and is a little lower maintenance for us. Gnome Keyring / libsecret is utilized and compatible with pretty well everything as well so that makes that end easy as well.
